tests/unit/lms/models/application_instance_test.py

```diff
@@ -75,6 +75,18 @@ def test_requesters_email_cant_be_null(self, db_session, application_instance):
         with pytest.raises(IntegrityError, match="requesters_email"):
             db_session.flush()
 
+    def test_lms_host(self, application_instance):
+        application_instance.lms_url = "https://example.com/lms/"
+
+        assert application_instance.lms_host() == "example.com"
```

Comment:
I assume we can always rely on this being fully qualified when we get it? As far as I can tell we don't actually enforce that. `urlparse` behavior is very iffy if you don't have a schema specified or implied with '//':

```python
for url in ("https://google.com:443/path", "google.com:443/path", "google.com"):
    print(url, '>>', urlparse(url).netloc or None)
    
# https://google.com:443/path >> google.com:443
# google.com:443/path >> None
# google.com >> None
```

Reply:
I'm pretty sure users can enter invalid values into the `/welcome` form for this! Even if they do enter a valid domain name that still doesn't mean that it's actually the correct domain name of their Canvas or Blackboard instance and will work when we try to use it to talk to the Canvas or Blackboard API. So we could do with better validation on the way in for sure

Reply:
I think we do prepend `https://` if the user doesn't enter a schema though

Reply:
https://github.com/hypothesis/lms/blob/28f158a9cb5edfaca254848581ed35edfbf5d242/lms/static/scripts/new-application-instance.js#L31-L42

Reply:
https://github.com/hypothesis/lms/blob/28f158a9cb5edfaca254848581ed35edfbf5d242/lms/static/scripts/new-application-instance.js#L1-L6

Reply:
There's no backend validation:

https://github.com/hypothesis/lms/blob/28f158a9cb5edfaca254848581ed35edfbf5d242/lms/views/application_instances.py#L27-L35